core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.spec.ts

```diff
@@ -247,6 +249,50 @@ describe('Translation Modal Component', () => {
     });
   });
 
+  describe('with a machine translatable translation language', () => {
+    beforeEach(fakeAsync(() => {
+      translationLanguageService.setActiveLanguageCode('es');
+      component.ngOnInit();
+      const sampleStateWiseContentMapping = {
+        stateName1: {contentId1: 'Please continue'}
+      };
+
+      const req = httpTestingController.expectOne(
+        '/gettranslatabletexthandler?exp_id=1&language_code=es');
+      expect(req.request.method).toEqual('GET');
+      req.flush({
+        state_names_to_content_id_mapping: sampleStateWiseContentMapping,
+        version: 1
+      });
+      flushMicrotasks();
+    }));
+
+    it('should offer translation', () => {
+      expect(component.languageSupportsMachineTranslations).toBeTrue();
+    });
+
+    it('should not retrieve translation until dropdown chevron is clicked',
+      fakeAsync(() => {
+        expect(component.hideMachineTranslation).toBeTrue();
+        component.toggleMachineTranslation();
+        const sampleMachineTranslationResponse:
+        MachineTranslatedTextBackendDict = {
```

Comment:
Is it possible to indent this somehow to make it clearer it's the declared type of `sampleMachineTranslationResponse`?